src/main/java/com/heatheryou/httpserver/ResponseBuilder.java

```diff
@@ -12,43 +12,39 @@
     public Response buildResponse(int statusCode, String[] entityHeaders, String content) {
         setStatusLine(statusCode);
         setEntityHeader(entityHeaders);
+        setHeader();
         setBody(content);
         String header = getHeader();
         String body = getBody();
         return new Response(header, body);
     }
 
-    public void setHeader() {
-        header = getStatusLine() + getEntityHeader();
+    private void setStatusLine(int statusCode) {
+        statusLine = StatusLine.HTTP_VERSION + CharacterSet.SPACE + statusCode + CharacterSet.SPACE +
+                StatusLine.getReasonPhrase(statusCode) + CharacterSet.CRLF;
     }
 
-    public String getHeader() {
-        setHeader();
-        return header;
+    private void setEntityHeader(String[] entityHeaders) {
+        entityHeader = String.join("", entityHeaders);
     }
 
-    public void setStatusLine(int statusCode) {
-        statusLine = StatusLine.HTTP_VERSION + CharacterSet.SPACE + statusCode + CharacterSet.SPACE +
-                StatusLine.getReasonPhrase(statusCode) + CharacterSet.CRLF;
+    private void setHeader() {
+        header = getStatusLine() + getEntityHeader();
     }
 
-    public String getStatusLine() {
-        return statusLine;
+    private void setBody(String content) {
+        body = content + CharacterSet.CRLF;
     }
 
-    public void setEntityHeader(String[] entityHeaders) {
-        entityHeader = String.join("", entityHeaders);
+    public String getStatusLine() {
+        return statusLine;
     }
 
     public String getEntityHeader() {
         return entityHeader;
     }
 
-    public void setBody(String content) {
-        body = content + CharacterSet.CRLF;
-    }
+    public String getHeader() { return header; }
 
-    public String getBody() {
-        return body;
-    }
+    public String getBody() { return body; }
```

Comment:
I really like this object abstraction of the various "parts" of the document. Even if it becomes evident later on that this is not the best way to express this, you've built a domain language in your code and already have a decent seam to start making changes.